src/camera_session.cpp

```diff
@@ -29,8 +29,19 @@ using namespace libcamera;
 
 CameraSession::CameraSession(const CameraManager* const cm) {
   PRINT_FUNC();
-  if (opts.camera < cm->cameras().size()) {
+  if (opts.camera < 0) {
+    for (size_t i = 0; i < cm->cameras().size(); ++i) {
+      camera_ = cm->cameras()[i];
```

Comment:
This doesn't account for the case where you for loop through everything and no camera was validated.